index.js

```diff
@@ -39,30 +40,100 @@ function getTorrentMetaBuffer () { // TODO Fix parsing bug so this can be remove
 }
 
 function getFile (fpath) {
-  return _getFile(this, fpath)
+  return _getFile(module.exports, fpath)
+  .then(file => {
+    debug('FILE path=' + (file || {}).path, 'found=' + (file != null), file)
+    return file
+  })
+}
+
+function fetch (req, opts) {
+  opts = opts || {}
+  let inject = false
+  let url = null
+
+  // Convert a FetchEvent to Request
+  if (global.FetchEvent && req instanceof global.FetchEvent) {
+    inject = req.clientId == null
```

Comment:
Why is the `clientId == null`? Is it when the request is resent by the service worker?